test/test_irods_http_api.py

```diff
@@ -2412,6 +2412,203 @@ def test_parallel_write_init_operation_returns_an_error_when_targeting_child_res
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            # The data object was closed successfully.
+            # Do not invoke the parallel_write_shutdown operation in the finally block.
+            invoke_parallel_write_shutdown = False
+
+            # Show the data was written to the iRODS server safely.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'calculate_checksum',
+                'lpath': data_object,
+                'replica-number': 1
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertEqual(result['checksum'], 'sha2:YS0DmQW7u51PaFVYl/iTsHFOctKtP1MZZ8JLnBwCLMI=')
+            replica_1_checksum = result['checksum']
+
+            # Show that each replica has a different checksum.
+            # This helps strengthen the test by proving the replication and parallel
+            # write did what was expected.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'calculate_checksum',
+                'lpath': data_object,
+                'replica-number': 0
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertEqual(result['checksum'], 'sha2:1SgRcbKcy3+4fjwMvf7xQNG5OZmiYzBVbNuMIgiWbBE=')
+            self.assertNotEqual(result['checksum'], replica_1_checksum)
+
+        finally:
+            # End the parallel write in case something failed.
+            # This avoids leaking of parallel-write resources in the HTTP API server.
+            if invoke_parallel_write_shutdown:
+                r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                    'op': 'parallel_write_shutdown',
+                    'parallel-write-handle': parallel_write_handle
+                })
+                self.logger.debug(r.content)
+
+            # Remove the data object.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'remove',
+                'lpath': data_object,
+                'catalog-only': 0,
+                'no-trash': 1
+            })
+            self.logger.debug(r.content)
+
+            # Remove the unixfilesystem resource.
+            r = requests.post(f'{self.url_base}/resources', headers=rodsadmin_headers, data={
+                'op': 'remove',
+                'name': ufs_resc
+            })
+            self.logger.debug(r.content)
+
+    def test_parallel_write_init_operation_returns_an_error_on_invalid_replica_number(self):
```

Comment:
Do we need tests for invalid inputs such as those added in https://github.com/irods/irods_client_http_api/commit/3b8a5b2cd79f7667dac7611b84e198fde3412252 (i.e. `test_targeting_replica_using_invalid_replica_number_or_resource_results_in_an_error`), or do those cover this case as well?

Reply:
Done.